index.bs

```diff
@@ -681,14 +706,574 @@ id=] must be the same as the [=window handle=].
+  UndefinedValue //
+  NullValue //
+  StringValue //
+  NumberValue //
+  BooleanValue //
+  BigIntValue //
+  SymbolValue //
+  ArrayValue //
+  FunctionValue //
+  ObjectValue //
+  FunctionValue //
+  RegExpValue //
+  DateValue //
+  MapValue //
+  SetValue //
+  WeakMapValue //
+  WeakSetValue //
+  IteratorValue //
+  GeneratorValue //
+  ErrorValue //
+  ProxyValue //
+  PromiseValue //
+  TypedArrayValue //
+  ArrayBufferValue //
+  NodeValue //
+  WindowProxyValue //
+}
+
+ObjectId = text;
+
+ListValue = [*RemoteValue];
+
+MappingValue = [*[(RemoteValue / text), RemoteValue]];
+
+UndefinedValue = {
+  type: "undefined",
+}
+
+NullValue = {
+  type: "null",
+}
+
+StringValue = {
+  type: "string",
+  value: text,
+}
+
+SpecialNumber = "NaN" / "-0" / "+Infinity" / "-Infinity";
+
+NumberValue = {
+  type: "number",
+  value: number / SpecialNumber,
+}
+
+BooleanValue = {
+  type: "boolean",
+  value: bool,
+}
+
+BigIntValue = {
+  type: "bigint",
+  value: text,
+}
+
+SymbolValue = {
+  type: "symbol",
+  objectId: ObjectId,
+}
+
+ArrayValue = {
+  type: "array",
+  objectId: ObjectId,
+  value?: ListValue,
+}
+
+ObjectValue = {
+  type: "object",
+  objectId: ObjectId,
+  value?: MappingValue,
+}
+
+FunctionValue = {
+  type: "function",
+  objectId: ObjectId,
+}
+
+RegExpValue = {
+  type: "regexp",
+  objectId: ObjectId,
+  value: text
+}
+
+DateValue = {
+  type: "date",
+  objectId: ObjectId,
+  value: text
+}
+
+MapValue = {
+  type: "map",
+  objectId: ObjectId,
+  value?: MappingValue,
+}
+
+SetValue = {
+  type: "set",
+  objectId: ObjectId,
+  value?: ListValue
+}
+
+WeakMapValue = {
+  type: "weakmap",
+  objectId: ObjectId,
+}
+
+WeakSetValue = {
+  type: "weakset",
+  objectId: ObjectId,
+}
+
+ErrorValue = {
+  type: "error",
+  objectId: ObjectId,
+}
+
+PromiseValue = {
+  type: "promise",
+  objectId: ObjectId,
+}
+
+TypedArrayValue = {
+  type: "typedarray",
+  objectId: ObjectId,
+}
+
+ArrayBufferValue = {
+  type: "arraybuffer",
+  objectId: ObjectId,
+}
+
+NodeValue = {
+  type: "node",
+  objectId: ObjectId,
+  value?: NodeProperties,
+}
+
+NodeProperties = {
+  nodeType: uint,
+  nodeValue: text,
+  localName?: text,
+  namespaceURI?: text,
+  childNodeCount: uint,
+  children?: [*NodeValue],
+  attributes?: {*text => text},
+  shadowRoot?: NodeValue / null,
+}
+
+WindowProxyValue = {
+  type: "window",
+  objectId: ObjectId,
+}
+```
+
+Issue: Add WASM types?
+
+Issue: Should WindowProxy get attributes in a similar style to Node?
+
+Issue: handle String / Number / etc. wrapper objects specially?
+
+<div algorithm>
+
+To <dfn>serialize as a remote value</dfn> given an |value|, a |max depth|,
+|node details|, and a |set of known objects|:
+
+  1. In the following list of conditions and associated steps, run the first set
+     of steps for which the associated condition is true:
+
+    <dl>
+      <dt>[=Type=](|value|) is Undefined
+      <dd>Let |remote value| be an [=object=] matching the <code>UndefinedValue</code>
```

Comment:
`[=object=]` is linking to https://w3c.github.io/FileAPI/#blob-url-entry-object

Should these be https://infra.spec.whatwg.org/#ordered-map instead?

Reply:
I changed things to use infra values now that we can serialize infra values to JSON.